packages/i18n-utils/src/localize-url.ts

```diff
@@ -132,3 +128,14 @@ export function localizeUrl( fullUrl: string, toLocale?: Locale ): string {
 	// Nothing needed to be changed, just return it unmodified.
 	return fullUrl;
 }
+
+export function useLocalizeUrl(): ( fullUrl: string, locale?: Locale ) => string {
+	const providerLocale = useLocale();
+
+	return ( fullUrl: string, locale?: Locale ) => {
+		if ( locale ) {
+			return localizeUrl( fullUrl, locale );
+		}
+		return localizeUrl( fullUrl, providerLocale );
```

Comment:
@p-jackson do you think useLocale() be called here instead of closing over the value above?

Reply:
You have to close over in the value in this case. Hooks are magic and can only be called from components or other hooks, and the lambda here isn't a hook.

Reply:
You might also want to wrap the return value in a `React.useCallback` to prevent un-necessary re-renders. This would guarantee that the values returned by `useLocalizeUrl()` will always referentially equal to each other for as long as the value of `providerLocale` doesn't change.

```ts
	return React.useCallback(
		( fullUrl: string, locale?: Locale ) => {
			if ( locale ) {
				return localizeUrl( fullUrl, locale );
			}
			return localizeUrl( fullUrl, providerLocale );
		},
		[ providerLocale ]
	);
```

Reply:
This might be premature optimisation though, not sure what the real-world scenario is that I'd be guarding against here ...